integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioSessionMigrIstioGateway.java

```diff
@@ -0,0 +1,460 @@
+import static oracle.weblogic.kubernetes.actions.TestActions.shutdownManagedServerUsingServerStartPolicy;
+import static oracle.weblogic.kubernetes.utils.ApplicationUtils.checkAppUsingHostHeader;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.checkServiceExists;
+import static oracle.weblogic.kubernetes.utils.ConfigMapUtils.createConfigMapAndVerify;
+import static oracle.weblogic.kubernetes.utils.DeployUtil.deployToClusterUsingRest;
+import static oracle.weblogic.kubernetes.utils.DomainUtils.createDomainAndVerify;
+import static oracle.weblogic.kubernetes.utils.FileUtils.generateFileFromTemplate;
+import static oracle.weblogic.kubernetes.utils.ImageUtils.createMiiImageAndVerify;
+import static oracle.weblogic.kubernetes.utils.ImageUtils.createOcirRepoSecret;
+import static oracle.weblogic.kubernetes.utils.ImageUtils.dockerLoginAndPushImageToRegistry;
+import static oracle.weblogic.kubernetes.utils.IstioUtils.createIstioDomainResource;
+import static oracle.weblogic.kubernetes.utils.IstioUtils.deployHttpIstioGatewayAndVirtualservice;
+import static oracle.weblogic.kubernetes.utils.IstioUtils.deployIstioDestinationRule;
+import static oracle.weblogic.kubernetes.utils.IstioUtils.getIstioHttpIngressPort;
+import static oracle.weblogic.kubernetes.utils.OperatorUtils.installAndVerifyOperator;
+import static oracle.weblogic.kubernetes.utils.PodUtils.checkPodDoesNotExist;
+import static oracle.weblogic.kubernetes.utils.PodUtils.checkPodReady;
+import static oracle.weblogic.kubernetes.utils.SecretUtils.createSecretWithUsernamePassword;
+import static oracle.weblogic.kubernetes.utils.ThreadSafeLogger.getLogger;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@DisplayName("Test WLS Session Migration via istio enabled using Istio gateway")
+@IntegrationTest
+class ItIstioSessionMigrIstioGateway {
+
+  private static String opNamespace = null;
+  private static String domainNamespace = null;
+
+  // constants for creating domain image using model in image
+  private static final String SESSMIGR_MODEL_FILE = "model.sessmigr.yaml";
+  private static final String SESSMIGR_IMAGE_NAME = "mii-image";
+
+  // constants for web service
+  private static final String SESSMIGR_APP_NAME = "sessmigr-app";
+  private static final String SESSMIGR_APP_WAR_NAME = "sessmigr-war";
+  private static final int SESSION_STATE = 4;
+  private static Map<String, String> httpAttrMap;
+
+  // constants for operator and WebLogic domain
+  private static String domainUid = "sessmigr-domain-1";
```

Comment:
Make the domain uid unique --> istio-gateway-sessionmigr

Reply:
Since all three test suites share the same yaml file, they have to use same domain name

Reply:
changed